src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/Analyzers/ConvertToLibraryImportFixer.cs

```diff
@@ -647,17 +651,25 @@ SyntaxNode WrapInvocationWithHRExceptionThrow(SyntaxNode node, SyntaxGenerator g
                                 argumentsToRemove.Add(attrArg);
                                 argumentsToAdd.Add(attrArg.WithExpression(
                                     SyntaxFactory.LiteralExpression(SyntaxKind.StringLiteralExpression,
-                                        SyntaxFactory.Literal(entryPoint + entryPointSuffix))));
+                                        SyntaxFactory.Literal(AppendSuffix(entryPoint, entryPointSuffix)))));
                             }
                         }
                         else
                         {
-                            argumentsToRemove.Add(attrArg);
-                            argumentsToAdd.Add(attrArg.WithExpression(
-                                SyntaxFactory.BinaryExpression(SyntaxKind.AddExpression,
-                                    attrArg.Expression,
-                                    SyntaxFactory.LiteralExpression(SyntaxKind.StringLiteralExpression,
-                                        SyntaxFactory.Literal(entryPointSuffix.ToString())))));
+                            string entryPointValue = methodSymbol.GetAttributes()
+                                .First(att => att.AttributeClass.ToDisplayString() == TypeNames.DllImportAttribute)?
+                                .NamedArguments.First(arg => arg.Key == nameof(DllImportAttribute.EntryPoint))
+                                .Value.Value as string;
```

Comment:
We can pull the entry-point name from the `dllImportData` object instead of going through the attribute data. It will be more efficient and cleaner.